CONTRIBUTING.md

```diff
@@ -0,0 +1,74 @@
+
+## Milestones
+We use milestones to track progress of releases. There are also 2 special milestones
+used for helping us keep work organized: `Upcoming - Minor` and `Upcoming - Major`
+
+`Upcoming - Minor` is used for keeping track of issues that aren't assigned to a specific
+release but could easily be addressed in a minor release. `Upcoming - Major` keeps track
+of issues that will need to be addressed in a major release. For example, if the current
+version is `2.2.0` an issue/PR could fall in to one of 4 different active milestones:
+`2.2.1`, `2.3.0`, `Upcoming - Minor`, or `Upcoming - Major`. If an issue pertains to a
+specific upcoming bug or minor release, it would go into `2.2.1` or `2.3.0`. If the issue/PR
+does not have a specific milestone yet, but it is likely that it will land in a `2.X` release,
+it should go into `Upcoming - Minor`. If the issue/PR is a large functionality add or change
+and/or it breaks compatibility, then it should be added to the `Upcoming - Major` milestone.
+An issue that we are not sure we will be doing will not be added to any milestone.
+
+A milestone (and hence release) is considered done when all outstanding issues/PRs have been closed or moved to another milestone.
+
+## Issues
+Issues are used as the primary method for tracking anything to do with the acs-engine project.
+
+### Issue Lifecycle
+The issue lifecycle is mainly driven by the core maintainers, but is good information for those
+contributing to acs-engine. All issue types follow the same general lifecycle. Differences are noted below.
+1. Issue creation
+2. Triage
+    - The maintainer in charge of triaging will apply the proper labels for the issue. This
+    includes labels for priority, type, and metadata (such as "orchestrator/k8s"). The only issue
+    priority we will be tracking is whether or not the issue is "critical." If additional
+    levels are needed in the future, we will add them.
+    - (If needed) Clean up the title to succinctly and clearly state the issue. Also ensure
+    that proposals are prefaced with "Proposal".
+    - Add the issue to the correct milestone. If any questions come up, don't worry about
+    adding the issue to a milestone until the questions are answered.
+    - We attempt to do this process at least once per work day.
+3. Discussion
+    - "Feature" and "Bug" issues should be connected to the PR that resolves it.
+    - Whoever is working on a "Feature" or "Bug" issue (whether a maintainer or someone from
+    the community), should either assign the issue to them self or make a comment in the issue
+    saying that they are taking it.
+    - "Proposal" and "Question" issues should stay open until resolved or if they have not been
+    active for more than 30 days. This will help keep the issue queue to a manageable size and
+    reduce noise. Should the issue need to stay open, the `keep open` label can be added.
+4. Issue closure
+
+## How to Contribute a Patch
```

Comment:
I think it'll be good to call this out in release process, which I'll start discussion on after this PR.